pkg/hashicorpdocs/locked.go

```diff
@@ -104,15 +104,37 @@ func containsSuggestionInHeader(doc *docs.Document) bool {
 
 	// Navigate through all table contents to look for suggestions.
 	for _, row := range t.TableRows {
+		if len(row.SuggestedDeletionIds) > 0 ||
+			len(row.SuggestedInsertionIds) > 0 ||
+			len(row.SuggestedTableRowStyleChanges) > 0 {
+			// We found a suggestion.
+			return true
+		}
 		for _, cell := range row.TableCells {
+			if len(cell.SuggestedDeletionIds) > 0 ||
+				len(cell.SuggestedInsertionIds) > 0 ||
+				len(cell.SuggestedTableCellStyleChanges) > 0 {
+				return true
+			}
 			for _, content := range cell.Content {
 				if para := content.Paragraph; para != nil {
+					if len(para.SuggestedBulletChanges) > 0 ||
+						len(para.SuggestedParagraphStyleChanges) > 0 ||
+						len(para.SuggestedPositionedObjectIds) > 0 {
+						return true
+					}
 					for _, elem := range para.Elements {
+						if auto := elem.AutoText; auto != nil {
```

Comment:
Could we add a few comments about what is being checked here and what level are we in wrt to the header table?